### PR TITLE
Add initial database setup and migrations

### PR DESCRIPTION
**Pull request by randyrod**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+version: "3"
+services:
+  db:
+    image: postgres:11-alpine
```
Comment (pgrm) [inline docker-compose.yml]:
```suggestion
    image: postgres:12-alpine
```

File: src/Traces.Data/Migrations/20191122154400_InitialDb.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    public partial class InitialDb : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.CreateSequence(
+                name: "EntityFrameworkHiLoSequence",
+                incrementBy: 10);
+
+            migrationBuilder.CreateTable(
+                name: "Traces",
```
Comment (pgrm) [inline src/Traces.Data/Migrations/20191122154400_InitialDb.cs]: table names should be singular or?

File: src/Traces.Data/TracesDbContext.cs [hunk 3]
```diff
@@ -16,9 +16,19 @@ protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
         {
             if (!optionsBuilder.IsConfigured)
             {
-                // todo: Setup dbConnection string
-                optionsBuilder.UseNpgsql(string.Empty);
+                optionsBuilder.UseNpgsql(
```
Comment (pgrm) [inline src/Traces.Data/TracesDbContext.cs]: you don't need to do it here, you can set it up using `...ContextFactory` - check out the other projects we have for copy/paste examples

File: src/Traces.Data/TracesDbContext.cs [hunk 4]
```diff
@@ -16,9 +16,19 @@ protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
...
+                    "Host=127.0.0.1;Database=Traces;Username=OpenTraces;Password=OpenTraces.2019",
+                    npgSqlOption => npgSqlOption.UseNodaTime());
             }
         }
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            modelBuilder.Entity<Trace>()
```
Comment (pgrm) [inline src/Traces.Data/TracesDbContext.cs]: I think you are missing the query filter setup to filter by the account code in the cookie / token of the current user automatically

Comment (randyrod) [reply src/Traces.Data/TracesDbContext.cs]: Omitted on purpose for this PR. Will be added in subsequent PR.

**Pull request merged**